lib/logstash/inputs/lumberjack.rb

```diff
@@ -29,19 +29,37 @@ class LogStash::Inputs::Lumberjack < LogStash::Inputs::Base
   # SSL key passphrase to use.
   config :ssl_key_passphrase, :validate => :password
 
+  # The lumberjack input using a fixed thread pool to do the actual work and
+  # will accept a number of client in a queue, before starting to refuse new
+  # connection. This solve an issue when logstash-forwarder clients are 
+  # trying to connect to logstash which have a blocked pipeline and will 
+  # make logstash crash with an out of memory exception.
+  config :max_clients, :validate => :number, :default => 1000
```

Comment:
If this is just the size of the accept queue, then I think the name `max_clients` is a misnomer. 

I also wouldn't necessarily note the bug this setting fixes in the documentation.